scanpy/queries/_queries.py

```diff
@@ -0,0 +1,250 @@
+    org, attrs, host="www.ensembl.org", use_cache=False
+) -> pd.DataFrame:
+    """
+    Retrieve gene annotations from ensembl biomart.
+
+    Parameters
+    ----------
+    {doc_org}
+    attrs : `List[str]`
+        Attributes to query biomart for.
+    {doc_host}
+    {doc_use_cache}
+
+    Returns
+    -------
+    A `pd.DataFrame` containing annotations.
+
+    Examples
+    --------
+    Retrieve genes coordinates and chromosomes
+
+    >>> annot = sc.queries.biomart_annotations(
+            "hsapiens",
+            ["ensembl_gene_id", "start_position", "end_position", "chromosome_name"],
+        ).set_index("ensembl_gene_id")
+    >>> adata.var[annot.columns] = annot
+    """
+    return simple_query(org=org, attrs=attrs, host=host, use_cache=use_cache)
+
+
+@doc_params(doc_org=_doc_org, doc_host=_doc_host, doc_use_cache=_doc_use_cache)
+def gene_coordinates(
+    org,
+    gene_name,
+    gene_attr="external_gene_name",
+    chr_exclude=[],
+    host="www.ensembl.org",
+    use_cache=False,
+) -> pd.DataFrame:
+    """
+    Retrieve gene coordinates for specific organism through BioMart.
+
+    Parameters
+    ----------
+    {doc_org}
+    gene_name :
+        The gene symbol (e.g. "hgnc_symbol" for human) for which to retrieve
+        coordinates.
+    gene_attr : `str`, optional (default: "external_gene_name")
+        The biomart attribute the gene symbol should show up for.
+    chr_exclude :
+        A list of chromosomes to exclude from query.
+    {doc_host}
+    {doc_use_cache}
+
+    Returns
+    -------
+    A `pd.DataFrame` containing gene coordinates for the specified gene symbol.
+    """
+    res = simple_query(
+        org=org,
+        attrs=["chromosome_name", "start_position", "end_position"],
+        filters={gene_attr: gene_name},
+        host=host,
+        use_cache=use_cache,
+    )
+    return res[~res["chromosome_name"].isin(chr_exclude)]
+
+
+@doc_params(doc_org=_doc_org, doc_host=_doc_host, doc_use_cache=_doc_use_cache)
+def mitochondrial_genes(
+    org, attrname="external_gene_name", host="www.ensembl.org", use_cache=False
+) -> pd.DataFrame:
+    """\
+    Mitochondrial gene symbols for specific organism through BioMart.
+
+    Parameters
+    ----------
+    {doc_org}
+    attrname : `str`, optional (default: "external_gene_name")
+        Biomart attribute field to return. Possible values include
+        "external_gene_name", "ensembl_gene_id", "hgnc_symbol", "mgi_symbol",
+        and "zfin_id_symbol".
+    {doc_host}
+    {doc_use_cache}
+
+    Returns
+    -------
+    An `np.array` containing identifiers for mitochondrial genes.
+    """
+    return simple_query(
+        org,
+        attrs=[attrname],
+        filters={"chromosome_name": ["MT"]},
+        host=host,
+        use_cache=use_cache,
+    )
+
+
+@singledispatch
+def enrich(
+    container,
+    org: str = "hsapiens",
+    correction_method: str = "analytical",
+    gprofiler_kwargs: dict = {},
+) -> pd.DataFrame:
```

Comment:
You said you’ve gone with `None`, so this is `Optional[pd.DataFrame]`

Reply:
Oops, only updated the docs. 

Any chance you know of some tooling where I could just specify the types once and have them propagate between annotations and docs? Or maybe a linter that would warn me about mismatches?

Reply:
That’s what I’ve worked on a lot in the scanpydoc package.

Just delete them from the docs, they’ll be added automatically!